### ISSUE TITLE
Feature request: Apply billboard effect to collision node

### ISSUE THREAD

**Issue by el-dee** [panda3d/panda3d#695]
Currently camera effects, like billboard effect, are not or not properly applied to the collision to the collision node attached to a node (or the collision traverser ignores them).

A practical use is having user clickable labels or objets that always face the camera and stays up: when the rotation applied by the billboard effect becomes too large, the collision solid is no longer at the same place than the object.

See https://discourse.panda3d.org/t/billboard-effect-on-clickable-label/24516 for a more complete discussion on the subject and a simple code to demonstrate the problem.

**Comment by rdb** [comment]
For this to make sense, the collision system needs to know which camera to consider, noting that the same scene can be rendered from multiple cameras.  Are you suggesting it should be possible to pass an optional camera to `traverse()`?

**Comment by el-dee** [comment]
I didn't thought of the consequences, and indeed specifying the camera would be needed and that might be overkill for the intended result ( don't know enough of the inner working of the engine to know what would be the required changes).

On the other hand I just tried with what I think is a world-only billboard effect :

    box.setEffect(BillboardEffect.make((0, 0, 1), False, False, 0.0, base.cam, (0, 0, 0)))

and yet the collision node does not seem to be rotated by the effect (In that case the actual camera used to render the scene should not matter as we are only using in world nodes ?)

**Comment by rdb** [comment]
It's conceivable that the way the CollisionTraverser is written does not take into account RenderEffects with `adjust_transform`.  I would agree that this should be fixed, since it would also apply to eg. CompassEffect.

**Comment by el-dee** [comment]
I fully agree with you :) I updated the title and add a remark in the description
